cppcache/src/ThinClientRegion.cpp

```diff
@@ -3328,19 +3333,22 @@ bool ThinClientRegion::executeFunctionSH(
           LOGWARN("ThinClientRegion::executeFunctionSH: Unexpected Exception");
         }
 
-        for (std::vector<OnRegionFunctionExecution*>::iterator iter2 =
-                 feWorkers.begin();
-             iter2 != feWorkers.end(); ++iter2) {
-          OnRegionFunctionExecution* worker = *iter2;
-          delete worker;
+        if(abortError == GF_NOERR) {
+          abortError = err;
         }
-        LOGDEBUG(
-            "ThinClientRegion::executeFunctionSH: Cleaned out the workers ");
-
-        GfErrTypeToException("ExecuteOnRegion:", err);
       }
     }
+    LOGERROR("ThinClientRegion::executeFunctionSH worker count = %d", feWorkers.size());
+
     delete worker;
```

Comment:
Why not store shared_ptr of worker so that when removed from collection and reference descopes it auto deletes? Using delete directly in C++11 is poor form now.